src/game/rendering_graph_node.c

```diff
@@ -422,6 +436,12 @@ void geo_process_master_list_sub(struct GraphNodeMasterList *node) {
                 // Move to the next DisplayListNode.
                 currList = currList->next;
             }
+
+            if (currLayer == LAYER_CIRCLE_SHADOW_TRANSPARENT) {
+                gSPTexture(gDisplayListHead++, 0xFFFF, 0xFFFF, 0, G_TX_RENDERTILE, G_OFF);
+                gSPSetGeometryMode(gDisplayListHead++, G_LIGHTING | G_CULL_BACK);
+                gDPSetCombineMode(gDisplayListHead++, G_CC_SHADE, G_CC_SHADE);
```

Comment:
Don't need to set the combinemode here. Every material already sets it before.

Reply:
Not every material sets combine mode, counterexample https://github.com/HackerN64/HackerSM64/blob/master/actors/capswitch/model.inc.c#L170C11-L217

Combinemode should not leak